py3-jupyterhub-hmacauthenticator.yaml

```diff
@@ -2,34 +2,75 @@
 package:
   name: py3-jupyterhub-hmacauthenticator
   version: 1.0
-  epoch: 1
+  epoch: 2
   description: Dummy Authenticator for JupyterHub
   copyright:
     - license: BSD-3-Clause
   dependencies:
-    runtime:
-      - python-3
+    provider-priority: 0
+
+vars:
+  pypi-package: jupyterhub-hmacauthenticator
+  import: hmacauthenticator
+
+data:
+  - name: py-versions
+    items:
+      3.10: '310'
+      3.11: '311'
+      3.12: '312'
+      3.13: '313'
 
 environment:
   contents:
     packages:
-      - build-base
-      - busybox
-      - ca-certificates-bundle
-      - py3-setuptools
-      - python-3
-      - wolfi-base
+      - py3-supported-build-base
+      - py3-supported-hatchling
 
 pipeline:
   - uses: fetch
     with:
       expected-sha256: b2b220a5d3762dd43beb6e29ed1bd0e414b3de8ef8b2d1835b9986260a19c374
       uri: https://files.pythonhosted.org/packages/source/j/jupyterhub-hmacauthenticator/jupyterhub-hmacauthenticator-${{package.version}}.tar.gz
 
-  - name: Python Build
-    uses: python/build-wheel
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+      runtime:
+        - py${{range.key}}-jupyterhub
```

Comment:
pip shows that it has no runtime dependencies. 
```bash
$ uv pip show jupyterhub-hmacauthenticator
Name: jupyterhub-hmacauthenticator
Version: 1.0
Location: /Users/anurag.kumar/projects/uv-project/.venv/lib/python3.11/site-packages
Requires:
Required-by:
```

Reply:
This doesn't seem to be true. Without the `jupyterhub` runtime dep the import tests are failing:
```
2025/01/06 12:56:10 WARN Traceback (most recent call last): uses=python/import
2025/01/06 12:56:10 WARN   File "<string>", line 1, in <module> uses=python/import
2025/01/06 12:56:10 WARN     import hmacauthenticator uses=python/import
2025/01/06 12:56:10 WARN   File "/usr/lib/python3.13/site-packages/hmacauthenticator/__init__.py", line 4, in <module> uses=python/import
2025/01/06 12:56:10 WARN     from jupyterhub.auth import Authenticator uses=python/import
2025/01/06 12:56:10 WARN ModuleNotFoundError: No module named 'jupyterhub' uses=python/import
```